File: Semana 11/Services/SistemaService.cs
using System;
using System.Collections.Generic;
using System.Net.Http.Json;
using System.Net.Http;
using Semana11.Models;

namespace Semana11.Services
{
    public class SistemaService
    {
        readonly HttpClient _httpClient;

        public SistemaService()
        {
            _httpClient = new HttpClient();
            _httpClient.BaseAddress = new Uri("http://localhost:5204/Sistemas");
        }

        public List<Sistema> Listar()
        {            
            //Início do trecho para alteração
            var response = _httpClient.GetFromJsonAsync<List<Sistema>>("").Result; 
            //Fim do trecho para alteração

            if (response == null)           
                throw new Exception("Impossível consultar Sistemas");

            return response;            
        }

        public Sistema? Pesquisar(int codigo)
        {
            //Início do trecho para alteração
            var response = _httpClient.GetFromJsonAsync<Sistema>($"/{codigo}").Result;
            //Fim do trecho para alteração

            return response;            
        }

        public void Incluir(int codigo, string nome)
        {
            //Início do trecho para alteração        
            var sistema = new Sistema { Codigo = codigo, Nome = nome };
    
            var result = _httpClient.PostAsJsonAsync<Sistema>("", sistema).Result;
            //Fim do trecho para alteração

            result.EnsureSuccessStatusCode();            
        }
        
    }
}

File: Semana 18/exercicio/src/app/services/sistema.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Sistema } from '../models/sistema';

@Injectable({
  providedIn: 'root'
})
export class SistemaService {

  apiUrl : string = 'http://localhost:5204/Sistemas';

  constructor(private http : HttpClient) { }

  Pesquisar(codigo : number) : Observable<Sistema> {
    return this.http.get<Sistema>(`${this.apiUrl}/${codigo}`);
  }

  Listar() : Observable<Sistema[]> {
    //Início do trecho para alteração
    return this.http.get<Sistema[]>(this.apiUrl);
    //Fim do trecho para alteração
  }

  Incluir(sistema : Sistema) : Observable<any> {
    //Início do trecho para alteração
    return this.http.post(this.apiUrl, sistema);
    //Fim do trecho para alteração
  }

}
